spec/models/user_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+
+  describe 'validations' do
+    subject { build(:user) }
+
+    it { should validate_presence_of(:email) }
+    it { should validate_presence_of(:password) }
+    it { should validate_uniqueness_of(:email) }
+    it { should allow_value('user@gmail.com').for(:email) }
+    it { should_not allow_value('usergmail.com').for(:email) }
+
+    describe "length_password" do
+      it "should return error if length less than 8" do
+        user = build(:user, password: '1111')
```

Comment:
What do you think about shoulda-matchers? It'll be clearer. And length tests more straightforward.

Reply:
@volonterx yep, I'm absolutely okay with shoulda-matchers for that, but it doesn't work properly by some reason. I'm not sure why, maybe it's a problem of using custom serialiser, anyway I decided do not waste time on that and switch to next task. I think I could add TODO for that, but it's not very important in our current state.